src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1025,6 +1059,56 @@ private void ValidateCallerMemberNameAttribute(AttributeSyntax node, BindingDiag
             diagnostics.Add(node.Name.Location, useSiteInfo);
         }
 
+        private void ValidateCallerArgumentExpressionAttribute(AttributeSyntax node, CSharpAttributeData attribute, BindingDiagnosticBag diagnostics)
+        {
+            MessageID.IDS_FeatureCallerArgumentExpression.CheckFeatureAvailability(diagnostics, node);
```

Comment:
>MessageID.IDS_FeatureCallerArgumentExpression.CheckFeatureAvailability(diagnostics, node); [](start = 12, length = 90)

Do we really want to do this? Couldn't the code with the attribute be compiled before?

Reply:
It looks like the attribute is out there for a long time.

---
In reply to: [600749166](https://github.com/dotnet/roslyn/pull/51952#discussion_r600749166) [](ancestors = 600749166)

Reply:
@AlekseyTs Hmm, so should we not restrict a language version for using the attribute? or find something in between (e.g, not bind it on earlier versions)

Reply:
> so should we not restrict a language version for using the attribute? or find something in between (e.g, not bind it on earlier versions)

We ususally follow the principle that binding doesn't change based on the language version. We might report an error in the process based on the language version. If this was a new attribute, I would be fine with complaining on the attribute application as well, but in this case (an existing attribute) I think we sould complain only when there is a behavior change about to happen. There is no behavior change around the attribute application itself.


---
In reply to: [600763555](https://github.com/dotnet/roslyn/pull/51952#discussion_r600763555) [](ancestors = 600763555)

Reply:
@AlekseyTs Not sure if I understand this correctly. Does that mean to report the error only on the callsite?

e.g:

```csharp
public void M(string x, [CallerArgumentExpression("x")] string y = null) { } // No error.

M(""); // Error.
M("", ""); // No error.
```

Reply:
> Not sure if I understand this correctly. Does that mean to report the error only on the callsite?

Yes, report an error only when we supply argument's value based on the attribute.

---
In reply to: [600776134](https://github.com/dotnet/roslyn/pull/51952#discussion_r600776134) [](ancestors = 600776134)

Reply:
Done.